tests/test_gui.py

```diff
@@ -52,6 +52,7 @@ def test_editor(mock_db, mock_ctx, qtbot):
 
     win = MainWindow(db_dir, False)
     win.show()
+    qtbot.addWidget(win)
```

Comment:
These changes seemed to make some test failures more deterministic and easier to understand. Registering widgets like this seems to be recommended in the pytest-qt tutorial: https://pytest-qt.readthedocs.io/en/latest/tutorial.html

I'm hoping that by adding this and fixing the tests, I might have got rid of the random segfaults after the test suite finishes, but I'm not particularly confident of that.

Reply:
I will miss our segfaults :pensive: